src/kxtj3/kxtj3.hpp

```diff
@@ -0,0 +1,579 @@
+   *
+   * LOW_RES valid only for ODR <= 200 Hz
+   *
+   * Be cautious not to set resolution to LOW_RES with the G_RANG being on 14-bits modes
+   *
+   * Sensor needs to be in standby mode when setting the sensor resolution
+   *
+   * @param resolution One of the KXTJ3_RESOLUTION_T values
+   * @throws std::runtime_error on failure
+   */
+  void SetResolution(KXTJ3_RESOLUTION_T resolution);
+
+  /**
+   * @brief Sets the ODR of the sensor
+   *
+   * Sensor needs to be in standby mode when setting the ODR
+   *
+   * @param odr One of the KXTJ3_ODR_T values
+   * @throws std::runtime_error on failure
+   */
+  void SetOdr(KXTJ3_ODR_T odr);
+
+  /**
+   * @brief Sets the ODR of the wakeup function of the sensor
+   *
+   * Sensor needs to be in standby mode when setting the ODR
+   *
+   * @param odr One of the KXTJ3_ODR_WAKEUP_T values
+   * @throws std::runtime_error on failure
+   */
+  void SetOdrForWakeup(KXTJ3_ODR_WAKEUP_T odr);
+
+  /**
+   * @brief Performs a self-test for digital communication of the sensor. The test sets DCST bit in 
+   * CTRL_REG2, then checks the DCST_RESP register for a 0xAA, after the register is read, its
+   * value is 0x55 and DCST bit is cleared.
+   *
+   * This method is called by SensorSelfTest also
+   *
+   * Sensor must be in standby mode before performing this action
+   *
+   * @throws std::runtime_error on failure
+   */
+  void SelfTestDigitalCommunication();
+
+  /**
+   * @brief Performs a self-test of the sensor. The test applies an electrostatic force to the sensor,
+   * simulating input acceleration. The test compares samples from all axis before and after
+   * applying the electrostatic force to the sensor. If the amount of acceleration increases according
+   * to the values defined in TABLE 1 of the datasheet (0.5 g), the test passes.
+   *
+   * The method prints out the values before and during test and the average difference for each axis
+   *
+   * See the datasheet for more information
+   *
+   * @throws std::runtime_error on failure
+   */
+  void SensorSelfTest();
+
+  /**
+   * @brief Performs a sensor software reset. The software reset clears the RAM of the sensor and resets all registers
+   * to pre-defined values.
+   *
+   * You should call kxtj3_sensor_init() after the software reset
+   *
+   * See the datasheet for more details
+   *
+   * @throws std::runtime_error on failure
+    */
+  void SensorSoftwareReset();
+
+  /**
+   * @brief Gets raw accelerometer data from the sensor
+   * 
+   * @param x Pointer to a floating point variable to store the x-axis value. Set to NULL if not wanted.
+   * @param y Pointer to a floating point variable to store the y-axis value. Set to NULL if not wanted.
+   * @param z Pointer to a floating point variable to store the z-axis value. Set to NULL if not wanted.
+   * @throws std::runtime_error on failure
+   */
+  void GetAccelerationRaw(float *x, float *y, float *z);
```

Comment:
Generally this should be by reference (not pointer), however,lets remove these altogether in favor of your methods which return vectors.